src/validation.cpp

```diff
@@ -295,28 +294,31 @@ void Chainstate::MaybeUpdateMempoolForReorg(
     AssertLockHeld(cs_main);
     AssertLockHeld(m_mempool->cs);
     std::vector<uint256> vHashUpdate;
-    // disconnectpool's insertion_order index sorts the entries from
-    // oldest to newest, but the oldest entry will be the last tx from the
+    // disconnectpool sorts the entries from
+    // oldest to newest. The oldest entry will be the last tx from the
     // latest mined block that was disconnected.
     // Iterate disconnectpool in reverse, so that we add transactions
     // back to the mempool starting with the earliest transaction that had
     // been previously seen in a block.
```

Comment:
Maybe I'm misunderstanding things, but I find this language quite unintuitive and if my understanding is correct - misleading. Assume we have a chaintip at height 100, and we reorged the last 2 blocks, i.e. 100 and 99 (in that order) are disconnected.
- "from oldest to newest": I would call a tx in block 99 older than a tx in block 100, but disconnectpool.take() returns txs from block 100 before block 99
- "the latest mined block that was disconnected" could mean (and this was my first intuition) "the latest block _that was disconnected_", i.e. block 99, or "the highest of the block(s) that were disconnected", i.e. the previous chaintip, which I think is what is meant here.

Suggested rephrasing, which (imo) removes that ambiguity and also highlights the intent a bit more:
```suggestion
    // disconnectpool.take() returns entries first sorted by descending blockheight (i.e. txs from
    // the previous chaintip appear first) and then by reversed in-block order.
    // For example, if blocks 100 and 99 are disconnected, disconnectpool.take() returns 
    // [100:n, 100:n-1, ..., 100:0, 99:m, 99:m-1, ..., 99:0]
    // Iterate disconnectpool.take(( in reverse, so that we ensure first adding parent transactions
    // before children are added (if any).
```

Actually, I think moving the first lines that describe the .take() return value to the `DisconnectedBlockTransactions::take` docstring makes probably more sense, and then just reference that here?